File: app/src/main/java/io/github/ovso/blackbox/utils/ObjectUtils.java
package io.github.ovso.blackbox.utils;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public class ObjectUtils {

    public static boolean isEmpty(Object object) {
        return object == null;
    }

    public static boolean isEmpty(Object[] array) {
        return array == null || array.length == 0;
    }

    public static boolean isEmpty(CharSequence str) {
        return (str == null || str.length() == 0);
    }

    @SuppressWarnings("rawtypes")
    public static boolean isEmpty(Collection collection) {
        return collection == null || collection.isEmpty();
    }

    @SuppressWarnings("rawtypes")
    public static boolean isEmpty(Map map) {
        return map == null || map.isEmpty();
    }

    public static boolean isEmpty(List<?> list) {
        return list == null || list.isEmpty();
    }
}

File: app/src/main/java/io/github/ovso/blackbox/utils/ObjectUtils.kt
package io.github.ovso.blackbox.utils

object ObjectUtils {

  fun isEmpty(`object`: Any?): Boolean {
    return `object` == null
  }

  fun isEmpty(array: Array<Any>?): Boolean {
    return array == null || array.size == 0
  }

  fun isEmpty(str: CharSequence?): Boolean {
    return str == null || str.length == 0
  }

  fun isEmpty(collection: Collection<*>?): Boolean {
    return collection == null || collection.isEmpty()
  }

  fun isEmpty(map: Map<*, *>?): Boolean {
    return map == null || map.isEmpty()
  }

  fun isEmpty(list: List<*>?): Boolean {
    return list == null || list.isEmpty()
  }
}
